src/Features/CSharp/Portable/AddBraces/CSharpAddBracesDiagnosticAnalyzer.cs

```diff
@@ -45,13 +45,7 @@ public void AnalyzeNode(SyntaxNodeAnalysisContext context)
             var statement = context.Node;
             var cancellationToken = context.CancellationToken;
 
-            var optionSet = context.Options.GetDocumentOptionSetAsync(statement.SyntaxTree, cancellationToken).GetAwaiter().GetResult();
-            if (optionSet == null)
-            {
-                return;
-            }
-
-            var option = optionSet.GetOption(CSharpCodeStyleOptions.PreferBraces);
+            var option = context.Options.GetOption(CSharpCodeStyleOptions.PreferBraces, statement.SyntaxTree, cancellationToken);
```

Comment:
This is nice! We can potentially go a step further and define extension methods on common context types and get the tree and cancellationToken from the context. For example:

```csharp
T GetOption<T>(this SyntaxNodeAnalysisContext context, IOption<T> option);
T GetOption<T>(this OperationAnalysisContext context, IOption<T> option);